src/libs/Trie/index.js

```diff
@@ -0,0 +1,86 @@
+import _ from 'underscore';
+import TrieNode from './TrieNode';
+
+class Trie {
+    constructor() {
+        this.root = new TrieNode();
+    }
+
+    /**
+    * Add a word to the Trie
+    * @param {String} word
+    * @param {Object} [metaData]
+    * @param {TrieNode} node
+    * @param {String} [containChar]
+    * @returns {void}
+    */
+    add(word, metaData, node = this.root, containChar) {
+        if (word.length === 0 && !containChar) {
+            throw new Error('Cannot insert empty word into Trie :', word);
+        }
+        if (word.length === 0) {
+            node.setCompleteWord();
+            node.setMetaData(metaData);
+            return;
+        } if (!node.children[word[0]]) {
+            // eslint-disable-next-line no-param-reassign
+            node.children[word[0]] = new TrieNode();
```

Comment:
Remove this comment too, handle it properly.
Are we fine with this @stitesExpensify?

Reply:
Same as above

Reply:
Done.